lib/still/compiler/incremental/node.ex

```diff
@@ -124,11 +124,23 @@ defmodule Still.Compiler.Incremental.Node do
         source_file: %Still.SourceFile{input_file: state.file, run_type: :compile}
       }
 
+      Logger.error(error)
       ErrorCache.set({:error, error})
+
+      if Application.get_env(:still, :compiling) do
+        System.stop(1)
+      end
+
       {:reply, :ok, state}
 
     :error, %PreprocessorError{} = e ->
+      Logger.error(e)
       ErrorCache.set({:error, e})
+
+      if Application.get_env(:still, :compiling) do
+        System.stop(1)
```

Comment:
Is there any way we can propagate this up? Feels a bit wrong to call `System` here. Also, we can further improve the logic by not setting the `ErrorCache` when compiling, right?

Reply:
I'm not sure I'm understanding the full context, but this feels like it should be a `raise` at some point, and not a `System.stop` which will turn out extremely unfriendly to the reader

(I understand this is multi-process, so it's not as straightforward as having a raise right here)

Reply:
The issue here is that during a `mix still.compile` we actually terminate the program. Raising will crash the process, but that's not what we want, and we can't move this responsibility to the supervisor because supervisors are meant to run forever. I didn't find anything that would help me there. I looked into it, and `System.stop` is a clean way to do it because it stops everything before it exists.

Reply:
Yeah, it can get messy quick. I'm fine with the current approach